src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -37,34 +29,31 @@ public override void Initialize()
             this.slotsManager = engine.SlotsManager;
             this.federationHistory = engine.FederationHistory;
             this.validator = engine.PoaHeaderValidator;
-            this.chainState = engine.ChainState;
-            this.network = this.Parent.Network;
-
-            this.maxReorg = this.network.Consensus.MaxReorgLength;
         }
 
         public override async Task RunAsync(RuleContext context)
         {
             ChainedHeader chainedHeader = context.ValidationContext.ChainedHeaderToValidate;
 
-            var header = chainedHeader.Header as PoABlockHeader;
+            // If we're evaluating a batch of received headers it's possible that we're so far beyond the current tip
+            // that we have not yet processed all the votes that may determine the federation make-up.
+            if (!this.federationHistory.CanGetFederationForBlock(chainedHeader))
+            {
+                // Mark header as insufficient to avoid banning the peer that presented it.
+                // When we advance consensus we will be able to validate it.
+                context.ValidationContext.InsufficientHeaderInformation = true;
 
-            PubKey pubKey = this.federationHistory.GetFederationMemberForBlock(context.ValidationContext.ChainedHeaderToValidate)?.PubKey;
+                this.Logger.LogWarning("The polls repository is too far behind to reliably determine the federation members.");
+                this.Logger.LogDebug("(-)[INVALID_SIGNATURE]");
+                PoAConsensusErrors.InvalidHeaderSignature.Throw();
```

Comment:
it's not invalid, it's just we can't validate it.

Reply:
THis behavior will stop header validation and ban the node that supplied that header.

Reply:
If you look a bit further down this code is not new and is not unique to this PR. The difference is that we have a special case for when we "know" the voting information is incomplete versus when we know it's not incomplete. When we know its incomplete we still need to throw a consensus error and NOT ban the peer (there is another change in this PR to take care of that). If we know its not incomplete then we throw a consensus error AND ban the peer. The purpose of setting `context.ValidationContext.InsufficientHeaderInformation = true;` is to avoid banning the peer. See https://github.com/stratisproject/StratisFullNode/pull/574/files#diff-ccf9726d55ab3b4d2051b536125a6ff2e4d5244ebc7583fa4289d22b01aa7815R908-R913.